MvvmCross.Forms/Presenters/MvxFormsPagePresenter.cs

```diff
@@ -958,5 +962,24 @@ public virtual void ReplacePageRoot(Page existingPage, Page page, MvxPagePresent
                 throw new MvxException(ex, "Cannot replace MainPage root");
             }
         }
+
+        protected virtual Task<bool> FindAndCloseViewFromViewModel(IMvxViewModel mvxViewModel, Page rootPage, MvxPagePresentationAttribute attribute)
+        {
+            var root = TopNavigationPage();
+            Page pageToClose = null;
+
+            // finding the view from viewmodel in navigation stack
+            pageToClose = root?.Navigation?.NavigationStack?
+                .OfType<IMvxPage>()
+                .FirstOrDefault(x => x.ViewModel == mvxViewModel) as Page;
```

Comment:
Wouldn't it make more sense to use`LastOrDefault` here?

Reply:
`LastOrDefault` will try to find the last occurrence of an item and in most cases will start from the beginning of a list which means it will always do `O(n)` since it needs to know if there is another items that matches the condition. while `FirstOrDefault` will stop once the condition is satisfied.